xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.h

```diff
@@ -21,6 +21,8 @@
 #include <memory>
 #include <vector>
 
+typedef struct ass_image ASS_Image;
```

Comment:
Possible to use `using` instead of the typedef?

Reply:
sure, this works:
```
using ASS_Image = struct ass_image;
```

Reply:
I'm not sure if there is any opinion about changing a forward declaration though, the original header, `ass.h` has the following:
```
typedef struct ass_image {
<snip>
} ASS_Image;
```

ref: https://github.com/libass/libass/blob/master/libass/ass.h#L62-L83

Reply:
I was under the impression that usually the c++ "typedef" declaration using the `using` keyword would be preferred but honestly no hard request from my side.
Since this is already declared, wouldn't a simple forward declaration of `struct ASS_Image` solve the issue without having to redefine the typedef?

Reply:
when just using `struct ASS_Image`

```
In file included from /home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp:12:
/home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererUtil.h:18:26: error: conflicting declaration ‘typedef struct ass_image ASS_Image’
   18 | typedef struct ass_image ASS_Image;
      |                          ^~~~~~~~~
In file included from /home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp:10:
/home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.h:24:8: note: previous declaration as ‘struct ASS_Image’
   24 | struct ASS_Image;
      |        ^~~~~~~~~
In file included from /home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitlesLibass.h:17,
                 from /home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayLibass.h:12,
                 from /home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.cpp:18:
/usr/include/ass/ass.h:83:3: error: conflicting declaration ‘typedef struct ass_image ASS_Image’
   83 | } ASS_Image;
      |   ^~~~~~~~~
/home/lukas/Documents/git/xbmc/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRenderer.h:24:8: note: previous declaration as ‘struct ASS_Image’
   24 | struct ASS_Image;
      |        ^~~~~~~~~
```